api/grpcserver/grpcserver_test.go

```diff
@@ -1426,8 +1427,39 @@ func TestMeshService(t *testing.T) {
 	}
 }
 
+func TestTransactionServiceSubmitUnsync(t *testing.T) {
+	require := require.New(t)
+	grpcService := NewTransactionService(&networkMock, txAPI, txMempool, &IsSyncerMock{})
+	shutDown := launchServer(t, grpcService)
+	defer shutDown()
+
+	// start a client
+	addr := "localhost:" + strconv.Itoa(cfg.NewGrpcServerPort)
+
+	// Set up a connection to the server.
+	conn, err := grpc.Dial(addr, grpc.WithInsecure())
+	require.NoError(err)
+
+	defer func() { require.NoError(conn.Close()) }()
+	c := pb.NewTransactionServiceClient(conn)
+
+	serializedTx, err := types.InterfaceToBytes(globalTx)
+	require.NoError(err, "error serializing tx")
+	res, err := c.SubmitTransaction(
+		context.Background(),
+		&pb.SubmitTransactionRequest{Transaction: serializedTx},
+	)
+	require.Error(err)
+	require.Nil(res)
```

Comment:
After this, could you update the SyncerMock so that it returns true to `IsSynced()` and try to submit again, which should then work?

Reply:
Sure, I'll modify it accordingly